lib/broccoli/amd-shim.js

```diff
@@ -4,16 +4,17 @@ const stew = require('broccoli-stew');
 module.exports = function shimAmd(tree, nameMapping) {
   return stew.map(tree, (content, relativePath) => {
     let name = nameMapping[relativePath];
+    let sections = [
+      '(function(define){\n',
+      content,
+      '\n})((function(){ function newDefine(){ var args = Array.prototype.slice.call(arguments);',
+    ];
     if (name) {
-      return [
-        '(function(define){\n',
-        content,
-        '\n})((function(){ function newDefine(){ var args = Array.prototype.slice.call(arguments); args.unshift("',
-        name,
-        '"); return define.apply(null, args); }; newDefine.amd = true; return newDefine; })());',
-      ].join('');
-    } else {
-      return content;
+      sections.push(' args.unshift("');
+      sections.push(name);
+      sections.push('");');
     }
+    sections.push(' return define.apply(null, args); }; newDefine.amd = true; return newDefine; })());');
```

Comment:
We have intentionally avoided specifying `define.amd`, our loader does **not** implement the full spec behavior (which is what `define.amd` is trying to imply).

Reply:
I still keep the `newDefine` override convention that the existing AMD shim has. Did I break it in some way?